### ISSUE TITLE
Support multiple json files for Swagger 2.0

### ISSUE THREAD

**Issue by analogue** [striglia/pyramid_swagger#147]
Currently, we only support Swagger 2.0 specs that are wholly contained in a single ``swagger.json`` file.  As part of [bravado-core's refactoring to support recursive $refs] (https://github.com/Yelp/bravado-core/issues/35), serving up a swagger.json that references other json files also becomes possible. As such, clients will create http requests for other files that embedded $refs point to.

To support this feature, pyramid-swagger needs to be able to serve up all files that make up a spec and register the appropriate views.  See [register_api_doc_endpoints] (https://github.com/striglia/pyramid_swagger/blob/ae2de7770d2eced733490adbdf51cf1d895f92af/pyramid_swagger/__init__.py#L46)

**Comment by striglia** [comment]
Thanks Semir. Can you briefly describe the necessary change here? Presumably at least a bump to the bravado version, although we don't explicitly limit it in this library. Perhaps some documentation?

**Comment by analogue** [comment]
I've prototyped the change in this [branch](https://github.com/striglia/pyramid_swagger/compare/master...analogue:recursive_refs?expand=1) to verify the version of bravado-core (4.0.0) that supports recursive refs works as expected. An end-to-end test using the internal example_happyhour was successful. 

Since pyramid view callables can't be ``partials`` I used a rather ugly hack to deduce the json file name from the request's path.
